daisy_workflows/image_test/oslogin-ssh-master-tester.py

```diff
@@ -0,0 +1,221 @@
+  MasterExecuteInSshRetry(
+      OSADMINLOGIN_TESTER, [TESTER_SH, 'test_login', host_osadminlogin],
+      expect_fail=expect_fail)
+  MasterExecuteInSshRetry(
+      OSLOGIN_TESTER, [TESTER_SH, 'test_login_sudo', host_oslogin],
+      expect_fail=True)
+  MasterExecuteInSshRetry(
+      OSADMINLOGIN_TESTER, [TESTER_SH, 'test_login_sudo', host_osadminlogin],
+      expect_fail=expect_fail)
+
+
+def TestOsLogin(level):
+  key_oslogin, key_osadminlogin = AddOsLoginKeys()
+  user_oslogin = GetServiceAccountUsername(OSLOGIN_TESTER)
+  user_osadminlogin = GetServiceAccountUsername(OSADMINLOGIN_TESTER)
+  SetEnableOsLogin(True, level)
+  CheckNss(user_oslogin, user_osadminlogin)
+  CheckAuthorizedKeys(user_oslogin, key_oslogin)
+  CheckAuthorizedKeys(user_osadminlogin, key_osadminlogin)
+  TestLoginFromSlaves(user_oslogin, user_osadminlogin)
+  RemoveOsLoginKeys()
+  TestLoginFromSlaves(user_oslogin, user_osadminlogin, expect_fail=True)
+  key_oslogin, key_osadminlogin = AddOsLoginKeys()
+  TestLoginFromSlaves(user_oslogin, user_osadminlogin)
+  SetEnableOsLogin(None, level)
+  TestLoginFromSlaves(user_oslogin, user_osadminlogin, expect_fail=True)
+  CheckNss(user_oslogin, user_osadminlogin, expect_empty=True)
+  CheckAuthorizedKeys(user_oslogin, key_oslogin, expect_empty=True)
+  CheckAuthorizedKeys(user_osadminlogin, key_osadminlogin, expect_empty=True)
+  RemoveOsLoginKeys()
+
+
+def TestMetadataWithOsLogin(level):
+  tester_key = MD.AddSshKey(MM.SSH_KEYS, level)
+  MD.TestSshLogin(tester_key)
+  SetEnableOsLogin(True, level)
+  MD.TestSshLogin(tester_key, expect_fail=True)
+  SetEnableOsLogin(None, level)
+  MD.TestSshLogin(tester_key)
+  MD.RemoveSshKey(tester_key, MM.SSH_KEYS, level)
+  MD.TestSshLogin(tester_key, expect_fail=True)
+
+
+def TestOsLoginFalseInInstance():
+  tester_key = MD.AddSshKey(MM.SSH_KEYS, MM.INSTANCE_LEVEL)
+  MD.TestSshLogin(tester_key)
+  SetEnableOsLogin(True, MM.PROJECT_LEVEL)
+  MD.TestSshLogin(tester_key, expect_fail=True)
+  SetEnableOsLogin(False, MM.INSTANCE_LEVEL)
+  MD.TestSshLogin(tester_key)
+  SetEnableOsLogin(None, MM.INSTANCE_LEVEL)
+  MD.TestSshLogin(tester_key, expect_fail=True)
+  SetEnableOsLogin(None, MM.PROJECT_LEVEL)
+  MD.TestSshLogin(tester_key)
+  MD.RemoveSshKey(tester_key, MM.SSH_KEYS, MM.INSTANCE_LEVEL)
+  MD.TestSshLogin(tester_key, expect_fail=True)
+
+
+def GetCurrentUsername():
+  # TODO: replace gcloud usage by python CLI
+  _, username = utils.Execute(
+      ['gcloud', 'compute', 'os-login', 'describe-profile',
+      '--format', 'value(posixAccounts.username)'], capture_output=True)
+  return username.strip()
+
+
+def AddKeyOsLogin(key):
+  # TODO: replace gcloud usage by python CLI
+  utils.Execute(
+      ['gcloud', 'compute', 'os-login', 'ssh-keys', 'add', '--key-file', key])
+
+
+def RemoveKeyOsLogin(key):
+  # TODO: replace gcloud usage by python CLI
+  utils.Execute(
+      ['gcloud', 'compute', 'os-login', 'ssh-keys', 'remove', '--key-file',
+      key])
```

Comment:
When line wrapping a list, the second line should match up to the first character after the open bracket. Apply this comment to the rest of this file.